conjur/client.py

```diff
@@ -30,84 +32,33 @@ class Client():
     # The method signature is long but we want to explicitly control
-    # what paramteres are allowed
-    #pylint: disable=too-many-arguments,too-many-locals
+    # what parameters are allowed
+    # pylint: disable=too-many-arguments,too-many-locals
     def __init__(self,
-                 account=None,
-                 api_key=None,
-                 ca_bundle=None,
-                 debug=False,
                  http_debug=False,
-                 login_id=None,
-                 password=None,
-                 ssl_verify=True,
-                 url=None):
-
-        self._setup_logging(debug)
-
-        logging.info("Initializing configuration...")
-
-        self._login_id = login_id
+                 ssl_verify=True):
 
-        config = {
-            'url': url,
-            'account': account,
-            'ca_bundle': ca_bundle,
-        }
+        logging.info("Initializing configuration")
 
-        if not url or not login_id or (not password and not api_key):
-            logging.info("Not all expected variables were provided. " \
-                "Using conjurrc as credential store...")
-            try:
-                on_disk_config = dict(ApiConfig())
-
-                # We want to retain any overrides that the user provided from params
-                # but only if those values are valid
-                for field_name, field_value in config.items():
-                    if field_value:
-                        on_disk_config[field_name] = field_value
-                config = on_disk_config
-
-            except Exception as exc:
-                raise ConfigException(exc) from Exception
+        # logging.info("Using conjurrc as credential store...")
+        try:
+            loaded_config = dict(ApiConfig())
+        except Exception as exc:
+            raise ConfigException(exc) from Exception
 
         # We only want to override missing account info with "default"
         # if we can't find it anywhere else.
-        if config['account'] is None:
-            config['account'] = "default"
-
-        if api_key:
-            logging.info("Using API key from parameters...")
-            self._api = Api(api_key=api_key,
-                            http_debug=http_debug,
-                            login_id=login_id,
-                            ssl_verify=ssl_verify,
-                            **config)
-        elif password:
-            logging.info("Creating API key with login ID/password combo...")
-            self._api = Api(http_debug=http_debug,
-                            ssl_verify=ssl_verify,
-                            **config)
-            self._api.login(login_id, password)
-        else:
-            logging.info("Using API key with netrc credentials...")
-            self._api = Api(http_debug=http_debug,
-                            ssl_verify=ssl_verify,
-                            **config)
+        if loaded_config['account'] is None:
+            loaded_config['account'] = "default"
```

Comment:
https://www.tutorialspoint.com/python/dictionary_get.htm

Reply:
updated to `loaded_config.get('account', default = "default")`

Reply:
@sigalsax ?